### PR TITLE
Added validation errors

### PR DESCRIPTION
**Pull request by SidharthBansal**
As asked in #2556 by @jywarren 
Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [ ] code is in uniquely-named feature branch and has no merge conflicts
* [ ] PR is descriptively titled
* [ ] PR body includes `fixes #0000`-style reference to original issue #
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. 

Thanks!

### CODE REVIEW

File: app/models/node.rb [hunk 1]
```diff
@@ -851,6 +851,14 @@ def can_tag(tagname, user, errors = false)
...
     elsif tagname.split(':')[0] == 'redirect' && Node.where(slug: tagname.split(':')[1]).length <= 0
       errors ? I18n.t('node.page_does_not_exist') : false
+    elsif tagname.split(':')[0] == 'facebook'
+      errors ? "Only Oauth can create such tags" : false
+    elsif tagname.split(':')[0] == 'twitter'
+      errors ? "Only Oauth can create such tags" : false
+    elsif tagname.split(':')[0] == 'google'
+      errors ? "Only Oauth can create such tags" : false
+    elsif tagname.split(':')[0] == 'github'
+      errors ? "Only Oauth can create such tags" : false
```
Comment (seafr) [inline app/models/node.rb]: By using `["facebook", ...].include? tagname.split(':')[0]` instead we can squeeze 8 lines into just 2 and avoid much repetition. What do you think?

Comment (SidharthBansal) [reply app/models/node.rb]: @seafr I want to know whether we want the same error message for all 4 cases or different for these. If same message is OK. I will do the required patch.
Thanks

Comment (seafr) [reply app/models/node.rb]: Ok, I see your point. I thought this was the final version as the `in progress` label was removed.

Comment (SidharthBansal) [reply app/models/node.rb]: @jywarren what do you suggest?

Comment (jywarren) [reply app/models/node.rb]: Hm, i like @seafr's suggestion! Let's do it!

Comment (seafr) [reply app/models/node.rb]: @SidharthBansal I suggest moving all 4 options into 1 `elseif` as I said above. You can also customise the error message to something like: 
`"... create the #{tagname...[0]} tag"` 
if you want to make it different for each case.

Comment (SidharthBansal) [reply app/models/node.rb]: Ok, updating in a minute. Thanks both of you

**Comment by jywarren** [comment]
This looks good! Is it ready for merging? 👍 😄

**Comment by SidharthBansal** [comment]
Yeah

**Comment by jywarren** [comment]
I'm going to test CodeClimate integration by closing and reopening... let's see!

**Comment by jywarren** [comment]
@publiclab/reviewers - I tried enabling CodeClimate -- it's configured in `.codeclimate.yml` in the root directory. It's a little strict... what do folks think about having it on? I can turn it off if it seems too unfriendly.

**Comment by jywarren** [comment]
Maybe we should turn off some of the checks -- that way we can "ease in" more gradually. As files get cleaner, we can turn on more checks? https://github.com/publiclab/plots2/blob/master/.codeclimate.yml

**Comment by jywarren** [comment]
Also notice I was able to log into CodeClimate by clicking "Details" and then approve it there despite the issues it complained about. So we can work around this... my main concern is that it looks too intimidating to newcomers -- and I wish it offered more advice to people on how to improve their code, rather than just complaining about it!

**Comment by jywarren** [comment]
So now that I've approved it, to see the original error it complained about, click **Show all checks** and click on Details for CodeClimate.

**Comment by jywarren** [comment]
OK, merging!! 😀

**Pull request merged**